### PR TITLE
updated contibuting.md

### PR DESCRIPTION
**Pull request by kozlov721**
Updated `CONTRIBUTING.md` with info about docs and some more details in the other sections.

Please take a look if you want to add something or if you think some parts are not all clear.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,26 +5,58 @@ It outlines our workflow and standards for contributing to this project.
...
+
+We use the [Epytext](https://epydoc.sourceforge.net/epytext.html) markup language for documentation.
+To verify that your documentation is formatted correctly, follow these steps:
+
+1. Download [`get-docs.py`](https://github.com/luxonis/python-api-analyzer-to-json/blob/main/gen-docs.py) script
+1. Run `python3 get-docs.py luxonis_ml` in the root directory.
+   - If the script runs successfully and produces `docs.json` file, your documentation is formatted correctly.
+   - **NOTE:** If the script fails, it might not give the specific error message. In that case, you can run
+     the script for each file individually until you find the one that is causing the error.
+
```
Comment (tersekmatija) [inline CONTRIBUTING.md:35]: Can we suggest something like a VSCode extension that might help you generate correct docs?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -5,26 +5,58 @@ It outlines our workflow and standards for contributing to this project.
...
 
-1. Tests are run automatically when you open a pull request.
+1. Checks and tests are run automatically when you open a pull request.
+1. For the tests to run, the [pre-commit](#pre-commit-hooks) hook must pass and
+   the [documentation](#documentation) must be built successfully.
 1. Review the GitHub Actions output if your PR fails.
-1. Fix any issues to ensure that both the pre-commit hooks and tests pass.
+1. Fix any issues to ensure that all checks and tests pass.
 
 ## Making and Reviewing Changes
```
Comment (tersekmatija) [inline CONTRIBUTING.md:62]: Maybe here we add a note that any relevant reviewers are also welcome?

Comment (tersekmatija) [reply CONTRIBUTING.md:62]: And if anything of significance is added, a test case for it must be added as well.

**Review by tersekmatija — APPROVED** [review]
Added some minor comments, looks good otherwise. Thanks!

File: CONTRIBUTING.md (hunk 1 above)
Comment (kozlov721) [reply CONTRIBUTING.md:35]: I wasn't able to find anything. The [autoDocstring](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring) extension doesn't support `epytext` (there's an open issue for it, but it's about 3 years old already). It should be supported in PyCharm though, if anyone is using it.

Comment (conorsim) [reply CONTRIBUTING.md:35]: @klemen1999 asked me about this and I found [this one](https://marketplace.visualstudio.com/items?itemName=AIC.docify) but haven't tried it

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -5,26 +5,60 @@ It outlines our workflow and standards for contributing to this project.
...
+- [Tests](#tests)
 - [GitHub Actions](#github-actions)
 - [Making and Reviewing Changes](#making-and-reviewing-changes)
 - [Notes](#notes)
 
-## Pre-commit Hook
+## Pre-commit Hooks
 
-We use a pre-commit hook to ensure code quality and consistency:
+We use a pre-commit hooks to ensure code quality and consistency:
```
Comment (conorsim) [inline CONTRIBUTING.md:17]:
```suggestion
We use pre-commit hooks to ensure code quality and consistency:
```

**Review by conorsim — APPROVED** [review]
LGTM

**Pull request merged**